### ISSUE TITLE
Contact names not persisted on prosody with mod_groups

### ISSUE THREAD

**Issue by enguerrand** [iNPUTmice/Conversations#4195]
Steps to reproduce:
* Edit contact details
* Set a custom name
* Disconnect and reconnect account
* The contact name is now reverted to the user fraction of the jid

The problem also occurs after some time without manually disconnecting the account but I am not sure which event is triggering the data loss in these cases.

I am using the fdroid version with a self hosted prosody.

**Comment by enguerrand** [comment]
Some additional information. In order to isolate the issue I tested the following:

* Disconnect Conversations
* Connect via another xmpp client (in this case gajim on the laptop)
* Rename a contact in gajim (manage contact -> Rename...)
* Disconnect gajim
* Reconnect Conversations
* Now the contact is renamed as wanted in Conversations, even after further disconnect / reconnect cycles

So to me it looks like the problem occurs while Conversations is saving the rename operation.

For the above test I enabled debug logging on the server but could not find any obvious differences between the Conversations' and gajim's write operation.

**Comment by licaon-kter** [comment]
Can you make an account on conversations.im and test with that?

**Comment by enguerrand** [comment]
Oh wait. I can narrow this down even further: I am using prosody's groups module: https://prosody.im/doc/modules/mod_groups

The problem only occurs with contacts in the same group as myself.

**Comment by licaon-kter** [comment]
Which server software? Did you use any guides?

**Comment by enguerrand** [comment]
prosody, using their docs. currently running on debian 10.
I'll probably update to bullseye soon. Let me check if that makes the problem go away and report back then.

**Comment by licaon-kter** [comment]
@enguerrand Join and ask in: xmpp:prosody@conference.prosody.im

**Comment by enguerrand** [comment]
Ok, if the problem persists after upgrading to bullseye I'll do that and report back. Thanks!

**Comment by enguerrand** [comment]
Oh boy. This is a premium example for an rtfm indeed. I looked at [the module's documentation](https://prosody.im/doc/modules/mod_groups) as a first step before posting to the room you recommended. And sure enough, it is stated there: 

> Currently the shared groups can only be modified by changing the configuration. Any changes the user makes to the shared group contacts in their client will be lost the next time they log in. 

I was really mislead by the fact that it seemed at first as if I could modify the contact's name using gajim. Turns out however that later on even that change got lost again.

Either way. Sorry for the noise and thanks again for taking the time to review my (unjustified) "bug report". I'll go ahead and close this issue.
